starknet_devnet/starknet_wrapper.py

```diff
@@ -547,6 +547,21 @@ async def load_messaging_contract_in_l1(
             self.starknet, network_url, contract_address, network_id
         )
 
+    async def mock_message_to_l2(self, call: CallL1Handler) -> dict:
+        """Handles L1 to L2 message mock endpoint"""
+
+        state = self.get_state()
+
+        # Execute transactions inside StarknetWrapper
+        async with self.__get_transaction_handler() as tx_handler:
+            tx_handler.internal_tx = call
```

Comment:
But how does it work with an external tx? All other tx handler use an internal counterpart (there's always a call to `.from_external`)

Reply:
fixed in https://github.com/Shard-Labs/starknet-devnet/pull/365/commits/8f1e7bb41cb060144d981dd6b122638b828b103f

Reply:
Conclusion - resolving this conversation and we can leave it as it is: we are parsing the request as `InternalL1Handler` in the endpoint and passing that to StarknetWrapper's method since InternalL1Handler doesn't have a corresponding external tx.